feat(metabase): add metabase

polarstack opened the pull request:
**Description**
Metabase sets up in five minutes, connecting to your database, and bringing its data to life in beautiful visualizations.
An intuitive interface makes data exploration feel like second nature—opening data up for everyone, not just analysts and developers.

⚒️ Fixes  # <!--(issue)-->

**⚙️ Type of change**

- [x] ⚙️ Feature/App addition
- [ ] 🪛 Bugfix
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🔃 Refactor of current code

**🧪 How Has This Been Tested?**
* Install Ubuntu 22.03
* Install k3s
* Install Helm
* git clone https://github.com/truecharts/charts.git
* cd ~/charts/charts/operators/cloudnative-pg/
* helm install cloudnative-pg .
* cd ~/charts/charts/operators/prometheus-operator
* helm install prometheus-operator .
* cd ~/charts/charts/incubator/metabase
* helm dep update
* k3s kubectl create ns metabase
* helm install -n metabase metabase .
* helm uninstall -n metabase metabase
* k3s kubectl delete ns metabase

**📃 Notes:**
<!-- Please enter any other relevant information here -->

**✔️ Checklist:**

- [x] ⚖️ My code follows the style guidelines of this project
- [x] 👀 I have performed a self-review of my own code
- [ ] #️⃣ I have commented my code, particularly in hard-to-understand areas
- [ ] 📄 I have made corresponding changes to the documentation
- [ ] ⚠️ My changes generate no new warnings
- [ ] 🧪 I have added tests to this description that prove my fix is effective or that my feature works
- [ ] ⬆️ I increased versions for any altered app according to semantic versioning

**➕ App addition**

If this PR is an app addition please make sure you have done the following.

- [ ] 🪞 I have opened a PR on [truecharts/containers](https://github.com/truecharts/containers) adding the container to TrueCharts mirror repo.
- [x] 🖼️ I have added an icon in the Chart's root directory called `icon.png`

---

_Please don't blindly check all the boxes. Read them and only check those that apply.
Those checkboxes are there for the reviewer to see what is this all about and
the status of this PR with a quick glance._

polarstack committed:
feat(metabase): add metabase

polarstack committed:
feat(metabase): add README.md

polarstack committed:
fix(metabase): image tag

polarstack committed:
feat(metabase): add plugins persistance and bump major

polarstack committed:
fix(metabase): fix questions yaml

xstar97 commented on charts/incubator/metabase/values.yaml:
```diff
@@ -0,0 +1,55 @@
+            startup:
+              type: http
+              path: /
+              port: "{{ .Values.service.main.ports.main.targetPort }}"
```
we can remove the configmap and and secrets as these can be set here instead.

```suggestion
workload:
  main:
    podSpec:
      containers:
        main:
          probes:
            liveness:
              type: http
              path: /
              port: "{{ .Values.service.main.ports.main.targetPort }}"
            readiness:
              type: http
              path: /
              port: "{{ .Values.service.main.ports.main.targetPort }}"
            startup:
              type: http
              path: /
              port: "{{ .Values.service.main.ports.main.targetPort }}"
          env:
            JAVA_TIMEZONE: "{{ .Values.TZ }}"
            MB_DB_TYPE: "postgres"
            MB_DB_USER: "{{ .Values.cnpg.main.user }}"
            MB_DB_DBNAME: "{{ .Values.cnpg.main.database }}"
            MB_DB_PORT: 5432
            MB_DB_HOST:
              secretKeyRef:
                name: cnpg-main-urls
                key: host
            MB_DB_PASS:
              secretKeyRef:
                name: cnpg-main-user
                key: password
```

xstar97 commented on charts/incubator/metabase/templates/common.yaml:
```diff
@@ -0,0 +1,15 @@
+  {{- $_ := set .Values.configmap "metabase-config" $configmap -}}
+{{- end -}}
+
+{{ include "tc.v1.common.loader.apply" . }}
```
```suggestion
{{ include "tc.v1.common.loader.all" . }}

```

xstar97 reviewed (changes requested):
just delete the configmap and secrets.tpl after making these changes

polarstack committed:
feat(metabase): move env to values.yaml

Co-authored-by: Xstar97TheNoob <9399967+xstar97@users.noreply.github.com>
Signed-off-by: polarstack <42521003+polarstack@users.noreply.github.com>

polarstack committed:
feat(metabase): remove renders from template

Co-authored-by: Xstar97TheNoob <9399967+xstar97@users.noreply.github.com>
Signed-off-by: polarstack <42521003+polarstack@users.noreply.github.com>

polarstack committed:
feat(metabase): delete templates

polarstack commented:
> just delete the configmap and secrets.tpl after making these changes

Hi @xstar97 
Added your changes to the pull request and deleted the templates. the idea was to offload config from values.yaml to remain as clean as possible. For an Version 2.0.0 I've identified roughly 33 environment variables for metabase which could improve the user experience during installation and configuration (e.g. mail and ldap config). So not sure at which point config and secret maps make sense..?

Anyway, from my perspective the PR is ready

xstar97 commented:
> > just delete the configmap and secrets.tpl after making these changes
> 
> Hi @xstar97 Added your changes to the pull request and deleted the templates. the idea was to offload config from values.yaml to remain as clean as possible. For an Version 2.0.0 I've identified roughly 33 environment variables for metabase which could improve the user experience during installation and configuration (e.g. mail and ldap config). So not sure at which point config and secret maps make sense..?
> 
> Anyway, from my perspective the PR is ready

Hmm really depends on the complexicity of those variables. But you create custom  section for those actually. Checkout nextcloud questions and values for that.

polarstack commented:
> > > just delete the configmap and secrets.tpl after making these changes
> > 
> > 
> > Hi @xstar97 Added your changes to the pull request and deleted the templates. the idea was to offload config from values.yaml to remain as clean as possible. For an Version 2.0.0 I've identified roughly 33 environment variables for metabase which could improve the user experience during installation and configuration (e.g. mail and ldap config). So not sure at which point config and secret maps make sense..?
> > Anyway, from my perspective the PR is ready
> 
> Hmm really depends on the complexicity of those variables. But you create custom section for those actually. Checkout nextcloud questions and values for that.

The vars are not complex per se, the issue might be that all these settings are configurable in the gui and stored in the DB. According the docs the var overwrites everything, but I was not able yet to test if it's then also get written in the DB correctly.

From my perspective it's good enough to start for 99% of the users

xstar97 commented on charts/incubator/metabase/values.yaml:
```diff
@@ -0,0 +1,61 @@
+image:
+  repository: tccr.io/truecharts/metabase
+  pullPolicy: IfNotPresent
+  tag: v0.46.7@sha256:96f48cfb662f80278fb4b0356a747eea4d171868c68c7dfec56acc9a24813e96
```
```suggestion
image:
  repository: tccr.io/truecharts/metabase
  pullPolicy: IfNotPresent
  tag: v0.47.0@sha256:3f82fcefeb7c41d74fd8fd42bf01e0f15defb3f3dcdbe756820f0e9a286e53c7
```

xstar97 commented on charts/incubator/metabase/Chart.yaml:
```diff
@@ -0,0 +1,27 @@
+apiVersion: v2
+appVersion: "0.46.7"
```
```suggestion
appVersion: "0.47.0"
```

xstar97 commented on charts/incubator/metabase/values.yaml:
```diff
@@ -0,0 +1,61 @@
+              path: /
+              port: "{{ .Values.service.main.ports.main.targetPort }}"
+            startup:
+              type: http
```
```suggestion
              type: tcp
```

xstar97 committed:
Update charts/incubator/metabase/values.yaml

Signed-off-by: Xstar97TheNoob <9399967+xstar97@users.noreply.github.com>

xstar97 committed:
Update charts/incubator/metabase/Chart.yaml

Signed-off-by: Xstar97TheNoob <9399967+xstar97@users.noreply.github.com>

xstar97 committed:
Update charts/incubator/metabase/values.yaml

Signed-off-by: Xstar97TheNoob <9399967+xstar97@users.noreply.github.com>

xstar97 commented on charts/incubator/metabase/Chart.yaml:
```diff
@@ -0,0 +1,27 @@
+  - https://github.com/truecharts/charts/tree/master/charts/incubator/metabase
+  - https://github.com/metabase/metabase
+type: application
+version: 1.0.0
```
all initial charts get added as 0.0.1
```suggestion
version: 0.0.1
```

xstar97 committed:
Update charts/incubator/metabase/Chart.yaml

Signed-off-by: Xstar97TheNoob <9399967+xstar97@users.noreply.github.com>

truecharts-admin commented:
This PR is locked to prevent necro-posting on closed PRs. Please create a issue or contact staff on discord if you want to further discuss this
